packages/tinacms/src/edit-state.tsx

```diff
@@ -38,6 +39,51 @@ export const TinaEditProvider = ({
   )
 }
 
+export const useTina = ({
+  query,
+  variables,
+  data,
+}: {
+  query: string
+  variables: object
+  data: object
+}) => {
+  const {
+    setRequest,
+    state,
+    isDummyContainer,
+    isLoading: contextLoading,
+  } = React.useContext(TinaDataContext)
+
+  const [waitForContextRerender, setWaitForContextRerender] = useState<boolean>(
+    !isDummyContainer
+  )
+
+  const isLoading = contextLoading || waitForContextRerender
+
+  React.useEffect(() => {
+    setRequest({ query, variables })
+  }, [JSON.stringify(variables), query])
+
+  // A bit of a hack here
+  // We need to wait 1 frame because the parent context will need to react to our
+  // new query that we've just sent when this hook was initialized.
+  // Otherwise, things get wonky when changing pages
+  useEffect(() => {
```

Comment:
So I wasn't super jazzed at needing to use this state, but I couldn't think of a workaround without significant work.

When you change pages, the frontend hook on the next page will fire its new request to the Context. It's going to take a frame for the new loading/payload state to be pushed down, so we wait a frame here. 
Once we allow multiple query requests (grouped by some type of id), we won't have to do this.